### PR TITLE
[RFC][UN-11823] Improve calendars and dropdowns in mobile

### PR DESCRIPTION
**Pull request by mmelo**
## JIRA Ticket

https://uniplaces.atlassian.net/browse/UN-11823

## Context

**TLDR: add on-demand native dropdown behaviour to uni-dropdown**

**Part 1 / 2** 
Implement behaviour on ember-cli-uniq to be used on SPA-Offer-Page's Full-edit mode Photo gallery.

The experience with calendars in mobile is not great, as they don't appear centered on the screen, and sometimes is not even possible to change months.
We need to fix this on the following calendars:

- Edit Blocked Period;
- Add New Blocked Period;

Also, we need to change the behaviour from the dropdown in the photo management (the units) to the native system.

## Screenshots

To be added.

### CODE REVIEW

**Review by henrmota — APPROVED** [review]

**Comment by tlfbrito** [comment]
@mmelo great work! 👍

**Review by tlfbrito — APPROVED** [review]

File: addon/templates/components/uni-dropdown.hbs [hunk 1]
```diff
@@ -1,27 +1,35 @@
-<div role="button" class="{{btnClass}} uni-dropdown__button {{errorClass}}" {{action "buttonClick"}}>
-  {{#if selected}}
-    {{#if selectedAlias}}
-      {{selectedAlias}}
+{{#if isNativeOnMobile}}
```
Comment (AndreJoaquim) [inline addon/templates/components/uni-dropdown.hbs]: This already exists: it's the `uni-select`

File: addon/templates/components/uni-dropdown.hbs [hunk 2]
```diff
@@ -1,27 +1,35 @@
...
+  <select class="{{btnClass}} uni-dropdown__button uni-dropdown__button--native-mobile {{errorClass}}" onchange={{action "optionClick" option=target.value}}>
```
Comment (AndreJoaquim) [inline addon/templates/components/uni-dropdown.hbs]: Couldn't you use `uni-select` here? Seems that it's really similar

**Review by AndreJoaquim — CHANGES_REQUESTED** [review]

File: addon/components/uni-dropdown.js [hunk 3]
```diff
@@ -21,7 +25,8 @@ export default Component.extend(ClickOutsideMixin, {
   buttonErrorClass: 'uni-dropdown__button--error',
   options: [],
   btnClass: '',
-  isNativeOnMobile: false,
+
+  isNativeOnMobile: computed.alias('media.isMobile'),
```
Comment (AndreJoaquim) [inline addon/components/uni-dropdown.js]: I think you can remove this, as it's no longer needed

File: addon/templates/components/uni-dropdown.hbs [hunk 4]
```diff
@@ -1,9 +1,5 @@
 {{#if isNativeOnMobile}}
```
Comment (AndreJoaquim) [inline addon/templates/components/uni-dropdown.hbs]: media.isMobile

**Review by AndreJoaquim — CHANGES_REQUESTED** [review]

**Review by AndreJoaquim — APPROVED** [review]

**Pull request merged**